src/applications/vaos/tests/e2e/appointment-list.cypress.spec.js

```diff
@@ -48,3 +49,296 @@ describe('VAOS appointment list', () => {
+      cy.get('.usa-alert-success').should('not.exist');
+      cy.get('.usa-alert-error').should('exist');
+    });
+  });
+
+  describe('upcoming appointments', () => {
+    it('should render upcoming appointments list', () => {
+      cy.findByText(/Your appointments/i).should('exist');
+      cy.get('[data-cy=upcoming-appointment-list-header]').should('exist');
+      cy.get('[data-cy=upcoming-appointment-list]').should('exist');
+      cy.get('#type-dropdown').should('exist');
+      cy.get('[data-cy=appointment-list-item]')
+        .first()
+        .should('exist');
+      cy.axeCheckBestPractice();
+    });
+
+    it('should navigate to upcoming appointment details', () => {
+      cy.get('[data-cy=appointment-list-item]')
+        .first()
+        .findByText(/Details/i)
+        .click();
+      cy.findByText(/Appointment detail/i).should('exist');
+    });
+  });
+
+  describe('requested appointments', () => {
+    beforeEach(() => {
+      cy.findByText(/Your appointments/i).should('exist');
+      cy.get('[data-cy=upcoming-appointment-list-header]').should('exist');
+      cy.get('[data-cy=upcoming-appointment-list]').should('exist');
+      cy.get('#type-dropdown')
+        .select('requested')
+        .should('have.value', 'requested');
+    });
+
+    it('should render requested appointments list', () => {
+      cy.get('[data-cy=requested-appointment-list]').should('exist');
+      cy.get('[data-cy=requested-appointment-list-item]')
+        .first()
+        .should('exist');
+    });
+
+    it('should navigate to requested appointment details', () => {
+      cy.get('[data-cy=requested-appointment-list-item]')
+        .first()
+        .findByText(/Details/i)
+        .click();
+      cy.findByText(/Request detail/i).should('exist');
+    });
+  });
+
+  describe('past appointments', () => {
+    beforeEach(() => {
+      cy.get('#type-dropdown')
+        .select('past')
+        .should('have.value', 'past');
+    });
+
+    it('should render past appointments list', () => {
+      cy.findByLabelText(/Select a date range/i).should('exist');
+      cy.get('[data-cy=past-appointment-list]').should('exist');
+      cy.get('[data-cy=past-appointment-list-header]').should('exist');
+      cy.get('[data-cy=appointment-list-item]')
+        .first()
+        .should('exist');
+      cy.axeCheckBestPractice();
+    });
+
+    it('should navigate to requested appointment details', () => {
+      cy.get('[data-cy=appointment-list-item]')
+        .first()
+        .findByText(/Details/i)
+        .click();
+      cy.findByText(/Request detail/i).should('exist');
+    });
+
+    /**
+     *  0 = Past 3 months
+     *  1 = Nov. YYYY - Jan. YYYY
+     *  2 = Aug. YYYY - Oct. YYYY
+     *  3 = May YYYY - July YYYY
+     *  4 = All of YYYY (current year)
+     *  5 = All of YYYY (previous year)
+     */
+    describe('date range selector', () => {
+      it('should select November through January', () => {
```

Comment:
The date ranges in the dropdown are dynamic, and it looks like if I set my system time a couple months ahead, a couple of these tests break because there are no appointments in some of the date ranges. You might look at the existing past appointments Cypress tests to see how/if that's handled there.

Reply:
@jbalboni I see two options that we could take: conditional checks or checking to just see if a H3 heading is present. The conditionals are a bit more involved and not sure the impact to time but would check for either the list or the no appointment alert. I do have all the data tags already so not to involved to add the conditionals

Reply:
Conditional checks like that are hard in Cypress, from what I remember. I think just checking for the h3 is ok, maybe in the future we can have the mock specifically generate an appointment in the right time range.